walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,85 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  private static int DAY_LENGTH = 24*60;
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    Set<String> attendees =  new HashSet<String>();
+    int meetingDur = (int)request.getDuration();
+    attendees.addAll(request.getAttendees());
+    if (meetingDur > DAY_LENGTH){
+      return possibleTimes;
+    }
+    if (events.isEmpty() || attendees.isEmpty()){
+      possibleTimes.add(TimeRange.WHOLE_DAY);
+      return possibleTimes;
+    }
+    /*A time is considered to be busy when an attendee is in an event & the meeting request*/
+    ArrayList<TimeRange> busyTimes = new ArrayList<TimeRange>();
+    for(Event event : events) {
+      Set<String> eventAttendees = event.getAttendees();
+      Set<String> overlap = new HashSet<>(eventAttendees);
+      overlap.retainAll(attendees);
+      if (!overlap.isEmpty()){
+        busyTimes.add(event.getWhen());
+      }
+    }
+  /*Combines overlapping busy times into an ArrayList*/
```

Comment:
same here, also in general -- these types of comments which describe an upcoming block of code, are often a good sign that this could benefit from a helper function.

Then ideally, instead of needing to describe what is happening, the method name can be descriptive enough to aid in understanding

Reply:
Got it